### PR TITLE
Package metadata.0.1.0

### PR DESCRIPTION
**Pull request by toots**
### `metadata.0.1.0`
Read metadata from various file formats
A pure OCaml library for reading files from various formats.

---
* Homepage: https://github.com/savonet/ocaml-metadata
* Source repo: git+https://github.com/savonet/ocaml-metadata.git
* Bug tracker: https://github.com/savonet/ocaml-metadata/issues

---
:camel: Pull-request generated by opam-publish v2.2.0

### CODE REVIEW

File: packages/metadata/metadata.0.1.0/opam [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+synopsis: "Read metadata from various file formats"
+description: "A pure OCaml library for reading files from various formats."
+maintainer: ["Samuel Mimram <smimram@gmail.com>"]
+authors: ["Samuel Mimram <smimram@gmail.com>"]
+license: "GPL-3.0"
+homepage: "https://github.com/savonet/ocaml-metadata"
+bug-reports: "https://github.com/savonet/ocaml-metadata/issues"
+depends: [
+  "dune" {>= "3.6"}
+  "ocaml" {>= "4.14.0"}
```
Comment (kit-ty-kate) [inline packages/metadata/metadata.0.1.0/opam:12]:
```suggestion
  "ocaml" {>= "4.14.0"}
  "conf-ffmpeg" {with-test}
```
```
#=== ERROR while compiling metadata.0.1.0 =====================================#
# context              2.2.0~alpha~dev | linux/x86_64 | ocaml-base-compiler.4.14.1 | pinned(https://github.com/savonet/ocaml-metadata/archive/refs/tags/v0.1.0.tar.gz)
# path                 ~/.opam/4.14/.opam-switch/build/metadata.0.1.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p metadata -j 127 @install @runtest
# exit-code            1
# env-file             ~/.opam/log/metadata-7-8da11f.env
# output-file          ~/.opam/log/metadata-7-8da11f.out
### output ###
# File "examples/dune", line 38, characters 3-9:
# 38 |    ffmpeg
#         ^^^^^^
# Error: Program ffmpeg not found in the tree or in PATH
#  (context: default)
# File "examples/dune", line 80, characters 3-9:
# 80 |    ffmpeg
#         ^^^^^^
# Error: Program ffmpeg not found in the tree or in PATH
#  (context: default)
# File "examples/dune", line 99, characters 3-9:
# 99 |    ffmpeg
#         ^^^^^^
# Error: Program ffmpeg not found in the tree or in PATH
#  (context: default)
# File "examples/dune", line 114, characters 3-9:
# 114 |    ffmpeg
#          ^^^^^^
# Error: Program ffmpeg not found in the tree or in PATH
#  (context: default)
# File "examples/dune", line 131, characters 3-9:
# 131 |    ffmpeg
#          ^^^^^^
# Error: Program ffmpeg not found in the tree or in PATH
#  (context: default)
# File "examples/dune", line 59, characters 3-9:
# 59 |    ffmpeg
#         ^^^^^^
# Error: Program ffmpeg not found in the tree or in PATH
#  (context: default)
```

File: packages/metadata/metadata.0.1.0/opam [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# This file is generated by dune, edit dune-project instead
+opam-version: "2.0"
+synopsis: "Read metadata from various file formats"
+description: "A pure OCaml library for reading files from various formats."
+maintainer: ["Samuel Mimram <smimram@gmail.com>"]
+authors: ["Samuel Mimram <smimram@gmail.com>"]
+license: "GPL-3.0"
```
Comment (kit-ty-kate) [inline packages/metadata/metadata.0.1.0/opam:7]:
```suggestion
license: "GPL-3.0-or-later"
```

**Comment by kit-ty-kate** [comment]
The tests fail with:
```
#=== ERROR while compiling metadata.0.1.0 =====================================#
# context              2.2.0~alpha~dev | linux/x86_64 | ocaml-base-compiler.5.0.0 | pinned(https://github.com/savonet/ocaml-metadata/archive/refs/tags/v0.1.0.tar.gz)
# path                 ~/.opam/5.0/.opam-switch/build/metadata.0.1.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p metadata -j 47 @install @runtest
# exit-code            1
# env-file             ~/.opam/log/metadata-7-c3df01.env
# output-file          ~/.opam/log/metadata-7-c3df01.out
### output ###
# File "examples/dune", line 95, characters 0-229:
#  95 | (rule
#  96 |  (target test.jpg)
#  97 |  (action
# .....
# 106 |    -frames:v
# 107 |    1
# 108 |    test.jpg)))
# (cd _build/default/examples && /usr/sbin/ffmpeg -f lavfi -i color=size=320x240:color=blue -vf 'drawtext=fontsize=30:fontcolor=white:x=(w-text_w)/2:y=(h-text_h)/2:text=Testing' -frames:v 1 test.jpg)
# ffmpeg version n5.1.2 Copyright (c) 2000-2022 the FFmpeg developers
#   built with gcc 12.2.0 (GCC)
#   configuration: --prefix=/usr --disable-debug --disable-static --disable-stripping --enable-amf --enable-avisynth --enable-cuda-llvm --enable-lto --enable-fontconfig --enable-gmp --enable-gnutls --enable-gpl --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libdav1d --enable-libdrm --enable-libfreetype --enable-libfribidi --enable-libgsm --enable-libiec61883 --enable-libjack --enable-libmfx --enable-libmodplug --enable-libmp3lame --enable-libopencore_amrnb --enable-libopencore_amrwb --enable-libopenjpeg --enable-libopus --enable-libpulse --enable-librav1e --enable-librsvg --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libsvtav1 --enable-libtheora --enable-libv4l2 --enable-libvidstab --enable-libvmaf --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxcb --enable-libxml2 --enable-libxvid --enable-libzimg --enable-nvdec --enable-nvenc --enable-opencl --enable-opengl --enable-shared --enable-version3 --enable-vulkan
#   libavutil      57. 28.100 / 57. 28.100
#   libavcodec     59. 37.100 / 59. 37.100
#   libavformat    59. 27.100 / 59. 27.100
#   libavdevice    59.  7.100 / 59.  7.100
#   libavfilter     8. 44.100 /  8. 44.100
#   libswscale      6.  7.100 /  6.  7.100
#   libswresample   4.  7.100 /  4.  7.100
#   libpostproc    56.  6.100 / 56.  6.100
# Input #0, lavfi, from 'color=size=320x240:color=blue':
#   Duration: N/A, start: 0.000000, bitrate: N/A
#   Stream #0:0: Video: rawvideo (I420 / 0x30323449), yuv420p, 320x240 [SAR 1:1 DAR 4:3], 25 tbr, 25 tbn
# Stream mapping:
#   Stream #0:0 -> #0:0 (rawvideo (native) -> mjpeg (native))
# Press [q] to stop, [?] for help
# [Parsed_drawtext_0 @ 0x56518bab3d00] Cannot find a valid font for the family Sans
# [AVFilterGraph @ 0x56518bab3ec0] Error initializing filter 'drawtext' with args 'fontsize=30:fontcolor=white:x=(w-text_w)/2:y=(h-text_h)/2:text=Testing'
# Error reinitializing filters!
# Failed to inject frame into filter network: No such file or directory
# Error while processing the decoded data for stream #0:0
# Conversion failed!
```
One other thing that is pretty annoying is that the `conf-ffmpeg` package installs the `ffmpeg` library on most platforms instead of the binary. I’m not sure what to do about that, I guess we have to creat a new `conf-ffmpeg-bin` package or something like that

**Comment by kit-ty-kate** [comment]
> Are there any guidelines as to what to do here?

I'm not sure what you need but if you need some tools to be installed you'd need to create new `conf-*` packages for them. If this is too complicated the tests should just either be simplified (e.g. one set of tests meant to be running in opam-repository and one set of tests meant to be running in your own CI) or simply disabled if it's too much work.

**Comment by toots** [comment]
Thanks @kit-ty-kate. I think that the solution is indeed to make the full test run optional and only trigger it inside our CI runs. I'll look into it.

**Comment by kit-ty-kate** [comment]
Thanks!

**Pull request merged**